microprofile/lra/jax-rs/src/main/java/io/helidon/microprofile/lra/NonJaxRsResource.java

```diff
@@ -202,7 +200,8 @@ private void sendResponse(ServerResponse res, Response response) {
         }
     }
 
-    void terminate(@Observes @BeforeDestroyed(ApplicationScoped.class) Object event) {
+   @PreDestroy
+    void terminate() {
```

Comment:
Why is this changed?

Reply:
In my local tests, I was seeing the class instantiated twice. From the stack traces in the debugger is seemed that CDI was instantiating it the second time in order to notify it of the before-destroyed event. Because the logic in the method was simply cleaning up resources allocated by the bean itself (and not really taking action related to the shut-down of other beans), using `@PreDestroy` not only resolved the problem I was seeing but also more closely aligned with the work the method was doing.

Reply:
@tomas-langer Are you OK with this? Was this your only concern? I would like to merge this PR.